src/webots/editor/WbProjectRelocationDialog.cpp

```diff
@@ -402,24 +402,24 @@ bool WbProjectRelocationDialog::validateLocation(QWidget *parent, QString &filen
     return false;
   }
 
-  // change filename parameter: get relative filename with respect to previous project path
+  // get relative filename with respect to previous project path
   QString absolutePath;
   if (mExternalProtoProjectPath.isEmpty())
     absolutePath = current->path();
   else
     absolutePath = mExternalProtoProjectPath;
-  filename = QDir(absolutePath).relativeFilePath(filename) + (filename.endsWith("/") ? "/" : "");
+  QString newFilename = QDir(absolutePath).relativeFilePath(filename) + (filename.endsWith("/") ? "/" : "");
```

Comment:
Previously this function was adjusting the filename and returning it to the caller.
Now this behavior changed and the caller will have a wrong and outdated filename.
Is this really wanted?

Reply:
`validateLocation` was changed to address this: https://github.com/cyberbotics/webots/pull/4860#discussion_r928663714
but as you pointed out this isn't a good solution either as it changes the behavior.
Then I don't think we can avoid having the intermediary `destination` variable, even if unused

Reply:
Yes, if it is needed then it is fine to have the `destination` variable.